src/js/hooks/productSupplier/form/useDeletePreferenceType.js

```diff
@@ -0,0 +1,97 @@
+import _ from 'lodash';
+import PropTypes from 'prop-types';
+import { useDispatch } from 'react-redux';
+
+import { hideSpinner, showSpinner } from 'actions';
+import productSupplierApi from 'api/services/ProductSupplierApi';
+import confirmationModal from 'utils/confirmationModalUtils';
+
+const useDeletePreferenceType = ({
+  preferenceTypeData,
+  afterDelete,
+  isDefaultPreferenceType,
+  onCancel,
+}) => {
+  const dispatch = useDispatch();
+
+  const isPreferenceTypeEmpty = !_.some(
+    Object.values(_.omit(preferenceTypeData || {}, 'id')),
+  );
```

Comment:
wouldn't it be cleaner to do it using `isEmpty`?
```js
const isPreferenceTypeEmpty = _.isEmpty(_.omit(preferenceTypeData || {}, 'id'))
```

Reply:
It's not the same.
As an empty preferenceType, we are considering { bidName: '', validityStartDate: '', ... } 
1. `isEmpty`
In this case, this function returns false because it checks if the object has any keys (it has, but with empty values)
2. `some` + `Object.values`
In this case, we are taking only the values of the keys, so it gives us an array like that ['', '', ...] and then we are checking the truthiness of this array, and because of the fact every value in this array is falsy, the whole array is also falsy.

Reply:
yep, my bad, didn't understand the context before your explanation.